From 803775fb4dd21a231421ed5215ebf6e962ca00d1 Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Mon, 17 Jun 2019 19:15:02 +0200
Subject: [PATCH] TRITON-1740 Sizing up flexible disk throws error but works
 nonetheless Reviewed by: Trent Mick <trentm@gmail.com> Approved by: Trent
 Mick <trentm@gmail.com>

---
 lib/do_instance/do_disk/do_resize.js | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/lib/do_instance/do_disk/do_resize.js b/lib/do_instance/do_disk/do_resize.js
index 6bf914c..3b849a0 100644
--- a/lib/do_instance/do_disk/do_resize.js
+++ b/lib/do_instance/do_disk/do_resize.js
@@ -67,6 +67,7 @@ function do_resize(subcmd, opts, args, cb) {
             if (size < ctx.disk.size && !opts.dangerous_allow_shrink) {
                 next(new Error('--dangerous-allow-shrink must be specified ' +
                                 'when shrinking a disk'));
+                return;
             }
 
             cli.tritonapi.resizeInstanceDisk({
@@ -74,9 +75,9 @@ function do_resize(subcmd, opts, args, cb) {
                 diskId: diskId,
                 size: size,
                 dangerousAllowShrink: opts.dangerous_allow_shrink
-            }, function onResized(err, res) {
+            }, function onResized(err, _, res) {
                 if (err) {
-                    cb(err);
+                    next(err);
                     return;
                 }
 
@@ -111,9 +112,10 @@ function do_resize(subcmd, opts, args, cb) {
                     console.log('Resized disk "%s" in %s', disk.id,
                                 common.humanDurationFromMs(duration));
                     next();
-                } else {
-                    next(new Error('Failed to resize disk'));
+                    return;
                 }
+
+                next(new Error('Failed to resize disk'));
             });
         }
     ]}, cb);
-- 
2.21.0

